src/components/AddRepoForm.js

```diff
@@ -5,22 +5,36 @@ import {CardPadding} from "../styles/Card";
 import {Flex} from "../styles/Grid";
 import api from "../lib/apiGraphQL";
 
+export function isValidRepoUrl(url) {
+  const regex = new RegExp("https?://github.com/", "i"); // passing "i" as 2nd argument to ignore case
+  const isFullUrl = regex.test(url);
+  const relativeRepoUrl = isFullUrl ? url.toLowerCase().replace(regex, "") : url.toLowerCase();
+
+  const [owner, repo] = relativeRepoUrl.split("/");
+  if (!owner || !repo) {
+    return [false, null];
+  }
+  return [true, relativeRepoUrl];
+}
+
```

Comment:
considering this doesn't have to be specific to this component. Can you move this function to a separate file into the lib folder?

Reply:
moved to lib/util.js. Hope this is fine.